util/LuaRules/Gadgets/ai_dbg_map.lua

```diff
@@ -83,23 +207,54 @@ function gadget:DrawWorldPreUnit()
 --		Spring.Echo(threatMap[(height - 1) * width + width])
 
 		if SYNCED.threatData.isDraw then
-			for x = 1, width do
-				px = (x - 1) * size
-				for z = 0, height - 1 do
-					value = threatMap[z * width + x] - base
-					if value > 0 then
-						pz = z * size
-						value = value / div
-						gl.Color(value, 0.0, 0.0, 0.6)
-						gl.DrawGroundQuad(px, pz, px + size, pz + size)
-					elseif value < 0 then
-						pz = z * size
-						value = -value / div
-						gl.Color(0.0, 0.0, value, 0.6)
-						gl.DrawGroundQuad(px, pz, px + size, pz + size)
+			-- I AM GOING TO DO THIS THE STUPID AND SLOW WAY, ideally, you 
+			
+			if threatMapChanged then 
+				threatMapChanged = false
+				clearInstanceTable(circleInstanceVBO) -- remove all our previous geometry from buffer
+				for x = 1, width do
+					px = (x - 1) * size
+					for z = 0, height - 1 do
+						value = threatMap[z * width + x] - base
+						local mycolor = {0.0, 0.0, 0.0, 0.0}
+						
+						if value > 0 then
+							pz = z * size
+							value = value / div
+							mycolor[1] = value
+							--gl.Color(value, 0.0, 0.0, 0.6)
+							--gl.DrawGroundQuad(px, pz, px + size, pz + size)
+						elseif value < 0 then
+							pz = z * size
+							value = -value / div
+							mycolor[3] = value
+							--gl.Color(0.0, 0.0, value, 0.6)
+							--gl.DrawGroundQuad(px, pz, px + size, pz + size)
+						end
+						pushElementInstance( -- pushElementInstance(iT,thisInstance, instanceID, updateExisting, noUpload, unitID)
```

Comment:
Note that I am pushing a (probably black) instance into the buffer here, when I really shouldnt youll see what i mean if this works)